Src/LinearSolvers/MLMG/AMReX_MLCurlCurl_K.H

```diff
@@ -0,0 +1,285 @@
+#ifndef AMREX_ML_CURL_CURL_K_H_
+#define AMREX_ML_CURL_CURL_K_H_
+#include <AMReX_Config.H>
+
+#include <AMReX_Array4.H>
+
+namespace amrex {
+
+/* Index types
+ * E_x        : (0,1,1)
+ * E_y        : (1,0,1)
+ * E_z        : (1,1,0)
+ * (curl E)_x : (1,0,0)
+ * (curl E)_y : (0,1,0)
+ * (curl E)_z : (0,0,1)
+ */
+
+AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+void mlcurlcurl_adotx_x (int i, int j, int k, Array4<Real> const& Ax,
```

Comment:
I feel like this "adotx" language is confusing given the multiple components. If I understand correctly, "adotx" comes from thinking about the operator as `A`, the input as `x` and the output as `Ax`. This can be confusing when `x` itself has x, y, and z components. Why not replace "adotx" with "apply" or something else?

Reply:
I use `A dot x` because linear equations are often written as `A x = b`. For example, https://www.mathworks.com/help/matlab/ref/mldivide.html